zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala

```diff
@@ -340,9 +340,10 @@ final case class Endpoint[PathInput, Input, Err, Output, Auth <: AuthType](
               case Some(HttpCodecError.CustomError("SchemaTransformationFailure", message))
                   if maybeUnauthedResponse.isDefined && message.endsWith(" auth required") =>
                 maybeUnauthedResponse.get
-              case Some(_) =>
+              case Some(HttpCodecError.MissingAuthorizationHeader) =>
+                Handler.succeed(Response.unauthorized)
+              case Some(error)                                     =>
```

Comment:
Not sure why it was not written this way, as it looks like in this case `asHttpCodecError` provides the same value as `cause.defects.head.asInstanceOf[HttpCodecError]`.